### PR TITLE
feature-khweb-2/create-and-implement-loaders

### PR DESCRIPTION
**Pull request by PietyE**
**Create large and medium loader** (https://jira.softserve.academy/browse/KHWEB-2)

- created two shared components for loaders
- implemented medium loaders to components with useEffect requests 
Screenshot from 2022-05-23 20-34-49
Screenshot from 2022-05-23 20-40-23

### CODE REVIEW

File: src/shared/components/Loaders/LoaderLarge/LoaderLarge.tsx [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+import { Box, CircularProgress } from '@mui/material';
+import './LoaderLarge.scss';
+
+export const LoaderLarge = () => (
+  <Box className="loader-large-container">
+    <CircularProgress color="primary" className="loader-large" size={500} />
```
Comment (topsart) [inline src/shared/components/Loaders/LoaderLarge/LoaderLarge.tsx:7]: Can you move size prop to sass?

File: src/shared/components/Loaders/LoaderMedium/LoaderMedium.tsx [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+import { Box, CircularProgress } from '@mui/material';
+import './LoaderMedium.scss';
+
+export const LoaderMedium = () => (
+  <Box className="loader-medium-container">
+    <CircularProgress color="primary" className="loader-medium" size={80} />
```
Comment (topsart) [inline src/shared/components/Loaders/LoaderMedium/LoaderMedium.tsx:7]: the same

File: src/shared/components/Loaders/LoaderMedium/LoaderMedium.tsx [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+  </Box>
```
Comment (topsart) [inline src/shared/components/Loaders/LoaderMedium/LoaderMedium.tsx:8]: the same

File: src/shared/components/Loaders/LoaderLarge/LoaderLarge.tsx (hunk 1 above)
Comment (kerrimov) [reply src/shared/components/Loaders/LoaderLarge/LoaderLarge.tsx]: no need for it. Use `size` props

File: src/shared/components/Loader/Loader.tsx [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+import { Box, CircularProgress } from '@mui/material';
+import { LoaderType } from './constants/constants';
+import './Loader.scss';
+
+interface LoaderProps {
+  type: string;
+}
+
+export const Loader: React.FC<LoaderProps> = ({ type }: LoaderProps) => {
+  const isLarge: boolean = type === LoaderType.LOADER_TYPE_LARGE;
```
Comment (kerrimov) [inline src/shared/components/Loader/Loader.tsx:11]: move isLarge to props and make it true by default

File: src/components/Footer/Footer.scss [hunk 5]
```diff
@@ -8,6 +8,7 @@
   min-height: 3rem;
   overflow: hidden;
   background: $secondary-main;
+  z-index: 5;
```
[the remark this answers is not in the document]
Comment (PietyE) [reply src/components/Footer/Footer.scss]: It ` s for loader isnt blur footer while loading, I changed it to 4 , it is enough

File: src/components/Slider/Slider.tsx [hunk 6]
```diff
@@ -9,15 +9,26 @@ import { ArrowDirection } from './constants/constants';
...
 import './Slider.scss';
 
 SwiperCore.use([Navigation, Autoplay]);
 
 export const Slider: React.FC = () => {
   const slides: Array<Slides> = useSelector(getSlides);
+  //should I create selector for it ?
+  const { isLoading } = useSelector<StoreState, SliderState>(
+    state => state.slider,
+  );
```
[the remark this answers is not in the document]
Comment (PietyE) [reply src/components/Slider/Slider.tsx]: Left so

File: src/shared/components/Loader/Loader.tsx (hunk 4 above)
Comment (PietyE) [reply src/shared/components/Loader/Loader.tsx]: Moved to default

File: src/shared/components/Loader/Loader.tsx [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+export const Loader: React.FC<LoaderProps> = ({
+  type,
+  isLarge = true,
+}: LoaderProps) => {
+  if (type !== LoaderType.LOADER_TYPE_LARGE) {
+    isLarge = false;
+  }
+  const className = isLarge ? 'large' : 'medium';
+  const size = isLarge ? 500 : 80;
```
[the remark this answers is not in the document]
Comment (PietyE) [reply src/shared/components/Loader/Loader.tsx:19]: moved

**Review by kerrimov — DISMISSED** [review]

**Review by kerrimov — APPROVED** [review]

**Pull request merged**